pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -466,18 +463,10 @@ func listEqualUnordered(a []string, b []string) bool {
 
 // update the node associated with the connection, after receiving a a packet from envoy, also adds the connection
 // to the tracking map.
-func (s *DiscoveryServer) initConnection(node *core.Node, con *XdsConnection) (func(), error) {
-	con.mu.RLock() // may not be needed - once per connection, but locking for consistency.
-	initialized := con.node != nil
-	con.mu.RUnlock()
-
-	if initialized {
```

Comment:
why remove this? seems like its useful

Reply:
method is called only if con.node != nil.

Reply:
But it would really help to have a comment.

Reply:
ah right. LGTM then